src/foam/box/SessionServerBox.java

```diff
@@ -187,9 +189,11 @@ public void send(X x, Box delegate, boolean authenticate, Message msg) {
       }
 
       // Sub context might have service override for the delegate
-      if ( effectiveContext instanceof SubX ) {
-        ((Skeleton) delegate).setDelegateObject(
-          effectiveContext.get(x.get(NSpec.class).getId()));
+      X last = effectiveContext;
```

Comment:
I don't think this is thread-safe (now or before). I'm continuing to examine the code and design.